pkg/registry/core/pod/storage/storage.go

```diff
@@ -223,3 +229,62 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 	// subresources should never allow create on update.
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation, false, options)
 }
+
+// EphemeralContainersREST implements the REST endpoint for adding EphemeralContainers
+type EphemeralContainersREST struct {
+	store *genericregistry.Store
+}
+
+var _ = rest.Updater(&EphemeralContainersREST{})
+
+// New creates a new EphemeralContainers resource
+func (r *EphemeralContainersREST) New() runtime.Object {
+	return &api.EphemeralContainers{}
+}
+
+// Update alters the EphemeralContainers field in PodSpec
+func (r *EphemeralContainersREST) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
+		return nil, false, errors.NewBadRequest("feature EphemeralContainers disabled")
+	}
+
+	obj, err := r.store.Get(ctx, name, &metav1.GetOptions{})
```

Comment:
we'll probably want this endpoint to work with patch/apply, which has a couple implications for this storage:
* it should implement Get() as well as Update()
* for Update(), the `old v1.Pod -> old v1.EphemeralContainers -> calling objInfo(oldEphemeralContainers) -> new v1.EphemeralContainers -> new v1.Pod` transformation should happen inside an UpdatedObjectInfo implementation you pass to the pod storage (and the old v1.Pod will be fed in, you won't need to look it up yourself). Something like this:

```go
	// Build an UpdatedObjectInfo to pass to the pod store.
	// It is given the currently stored v1.Pod and transforms it to the new pod that should be stored.
	updatedPodInfo := rest.DefaultUpdatedObjectInfo(nil, func(ctx context.Context, oldObject, _ runtime.Object) (newObject runtime.Object, err error) {
		oldPod, ok := oldObject.(*api.Pod)
		if !ok {
			return nil, fmt.Errorf("unexpected type %T", oldObject)
		}

		newEphemeralContainersObj, err := objInfo.UpdatedObject(ctx, ephemeralContainersInPod(oldPod))
		if err != nil {
			return nil, err
		}
		newEphemeralContainers, ok := newEphemeralContainersObj.(*api.EphemeralContainers)
		if !ok {
			return nil, fmt.Errorf("unexpected type %T", newEphemeralContainersObj)
		}

		// avoid mutating
		newPod := oldPod.DeepCopy()
		// identity, version (make sure we're working with the right object, instance, and version)
		newPod.Name = newEphemeralContainers.Name
		newPod.Namespace = newEphemeralContainers.Namespace
		newPod.UID = newEphemeralContainers.UID
		newPod.ResourceVersion = newEphemeralContainers.ResourceVersion
		// ephemeral containers
		newPod.Spec.EphemeralContainers = newEphemeralContainers.EphemeralContainers

		return newPod, nil
	})

	obj, _, err := r.store.Update(ctx, name, updatedPodInfo, createValidation, updateValidation, false, options)
	if err != nil {
		return nil, false, err
	}
	return ephemeralContainersInPod(obj.(*api.Pod)), false, err
```